apps/token-manager/app/src/App.js

```diff
@@ -86,100 +98,151 @@ class App extends React.PureComponent {
       appStateReady,
+      connectedAccount,
       groupMode,
       holders,
       isSyncing,
+      layoutName,
       maxAccountTokens,
       numData,
+      requestMenu,
+      theme,
       tokenAddress,
       tokenDecimalsBase,
       tokenName,
       tokenSupply,
       tokenSymbol,
       tokenTransfersEnabled,
-      connectedAccount,
-      requestMenu,
     } = this.props
+
     const { assignTokensConfig, sidepanelOpened } = this.state
 
     return (
-      <Main assetsUrl="./aragon-ui">
-        <div css="min-width: 320px">
-          <IdentityProvider
-            onResolve={this.handleResolveLocalIdentity}
-            onShowLocalIdentityModal={this.handleShowLocalIdentityModal}
-          >
-            <SyncIndicator visible={isSyncing} />
-            <AppLayout
-              title="Tokens"
-              afterTitle={tokenSymbol && <Badge.App>{tokenSymbol}</Badge.App>}
-              onMenuOpen={requestMenu}
-              mainButton={{
-                label: 'Add tokens',
-                icon: <AssignTokensIcon />,
-                onClick: this.handleLaunchAssignTokensNoHolder,
-              }}
-              smallViewPadding={0}
+      <IdentityProvider
+        onResolve={this.handleResolveLocalIdentity}
+        onShowLocalIdentityModal={this.handleShowLocalIdentityModal}
+      >
+        <SyncIndicator visible={isSyncing} />
+
+        <Header
+          primary={
+            <div
+              css={`
+                display: flex;
+                align-items: center;
+              `}
             >
-              {appStateReady && holders.length > 0 ? (
-                <Holders
-                  holders={holders}
-                  groupMode={groupMode}
-                  maxAccountTokens={maxAccountTokens}
-                  tokenAddress={tokenAddress}
-                  tokenDecimalsBase={tokenDecimalsBase}
-                  tokenName={tokenName}
-                  tokenSupply={tokenSupply}
-                  tokenSymbol={tokenSymbol}
-                  tokenTransfersEnabled={tokenTransfersEnabled}
-                  userAccount={connectedAccount}
-                  onAssignTokens={this.handleLaunchAssignTokens}
-                  onRemoveTokens={this.handleLaunchRemoveTokens}
-                />
-              ) : (
-                !isSyncing && (
-                  <EmptyState
-                    onActivate={this.handleLaunchAssignTokensNoHolder}
-                  />
-                )
-              )}
-            </AppLayout>
-            <SidePanel
-              title={
-                assignTokensConfig.mode === 'assign'
-                  ? 'Add tokens'
-                  : 'Remove tokens'
-              }
-              opened={sidepanelOpened}
-              onClose={this.handleSidepanelClose}
-              onTransitionEnd={this.handleSidepanelTransitionEnd}
+              <h1
+                css={`
+                  ${textStyle(layoutName === 'small' ? 'title3' : 'title2')};
```

Comment:
Hmm, is `layoutName === 'small'` the same as `below('medium')`? I like having more utils but maybe we should consistently only use one no? To avoid having to update two places in case something changes and having to maintain these in sync?

Reply:
On a second idea, the styles for the `Header` `primary` prop (`title2` & `title2`), shouldn't these be set at component level? You mentioned there should only be one `Header` as it is implemented so I'd assume it takes care of the styling too

Reply:
> Hmm, is layoutName === 'small' the same as below('medium')?

Not exactly, it also:

- Takes into consideration the spacing around a given layout.
- More importantly, can take another width than the viewport width (which makes it work on non-iframe apps too).

See the implementation: https://github.com/aragon/aragon-ui/blob/newstyle/src/components/Layout/Layout.js#L13-L24

> I like having more utils but maybe we should consistently only use one no? To avoid having to update two places in case something changes and having to maintain these in sync?

Yes the idea is to move to `Layout`, except when the actual viewport dimensions are needed.

>  On a second idea, the styles for the Header primary prop (title2 & title2), shouldn't these be set at component level? You mentioned there should only be one Header as it is implemented so I'd assume it takes care of the styling too

Yes it’s something that remains to be solved… [styles are applied](https://github.com/aragon/aragon-ui/blob/newstyle/src/components/Header/Header.js#L33-L44), except if a React node is passed, in which case we can’t apply the styles since we don’t control the component. There are different ways to address this (like applying default CSS styles, which might be surprising since we don’t do that anywhere else), but I wanted to let it some time before going for one solution or another.